cbackup.sh

```diff
@@ -578,15 +578,65 @@ function do_restore() {
     done
 }
 
-# Run action
-echo "Performing action '$action'"
-if [[ "$action" == "backup" ]]; then
-    do_backup
-elif [[ "$action" == "restore" ]]; then
-    do_restore
-else
-    die "Unknown action '$action'"
-fi
+function usage() {
+    echo -n "Usage: `basename $0` <ACTION> [OPTIONS]
+
+Actions:
+  backup        Perform a system backup
+  restore       Restore an existing backup
+  help          Show usage
+"
+}
+
+# Parse action and any additional options
+OPTIND=2
```

Comment:
Our code style convention is for constants to be `UPPER_CASE` and variables to be `snake_case`.

It's also not clear what this variable does at a glance from looking at its name.

Reply:
OPTIND is a variable for `getopts`. It specifies the index to start scanning for options at. If we remove this and leave it default (`OPTIND=1`), we scan for options after the basename. Ex: `cbackup [OPTIONS] <ACTION>`. With `OPTIND=2`, `getopts` uses this format: `cbackup <ACTION> [OPTIONS]`

Reply:
https://unix.stackexchange.com/questions/140840/using-getopts-to-parse-options-after-a-non-option-argument